src/github.com/getlantern/detour/whitelist.go

```diff
@@ -15,47 +14,56 @@ var (
 	whitelist   = make(map[string]wlEntry)
 )
 
-func InitWhitelist(wl map[string]time.Time) {
+// AddToWl adds a domain to whitelist, all subdomains of this domain
+// are also considered to be in the whitelist.
+func AddToWl(addr string, permanent bool) {
 	muWhitelist.Lock()
 	defer muWhitelist.Unlock()
-	for k, v := range wl {
-		whitelist[k] = wlEntry{true, v}
-	}
-	return
+	whitelist[addr] = wlEntry{permanent}
 }
 
-func DumpWhitelist() (wl map[string]time.Time) {
-	wl = make(map[string]time.Time)
+func RemoveFromWl(addr string) {
+	muWhitelist.Lock()
+	defer muWhitelist.Unlock()
+	delete(whitelist, addr)
+}
+
+func DumpWhitelist() (wl []string) {
+	wl = make([]string, 1)
 	muWhitelist.Lock()
 	defer muWhitelist.Unlock()
 	for k, v := range whitelist {
-		wl[k] = v.addTime
+		if v.permanent {
+			wl = append(wl, k)
+		}
 	}
 	return
 }
 
-func whitelisted(addr string) bool {
+func whitelisted(addr string) (in bool) {
 	muWhitelist.RLock()
 	defer muWhitelist.RUnlock()
-	_, in := whitelist[addr]
-	return in
+	for ; addr != ""; addr = getParentDomain(addr) {
```

Comment:
Not a biggie, but we could slightly optimize this through starting with checking the root domain and then adding subdomains since there are very few sites in the whitelist that include subdomains.

In practice, however, it's probably not a performance issue at all, and it would be slightly more complicated, so let's stick with what you've got!